### ISSUE TITLE
MaxListenersExceededWarning: Possible EventEmitter memory leak detected.

### ISSUE THREAD

**Issue by patchie** [huggingface/chat-ui#548]
Running dev, and no errors until i try to write into the chat interface on the website locally hosted in WSL2 (win11).

Worked before i updated to version v.0.6.0

error message in web ui:
![image](https://github.com/huggingface/chat-ui/assets/1792727/adc2f421-6cb7-400d-b559-1240b13ff349)

Error message in terminal:

> root@xxxxxxxxx:/mnt/c/WSL/HuggingChat test/AI# npm run dev-chat-ui
> 
> > ai@1.0.0 dev-chat-ui
> > cd ../chat-ui && npm run dev -- --host 0.0.0.0
> 
> 
> > chat-ui@0.6.0 dev
> > vite dev --host 0.0.0.0
> 
> 
> 
>   VITE v4.3.9  ready in 15775 ms
> 
>   ➜  Local:   http://localhost:5173/
>   ➜  Network: http://172.xx.142.227:5173/
>   ➜  press h to show help
> (node:80446) **MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [TLSSocket]. Use emitter.setMaxListeners() to increase limit**
> (Use `node --trace-warnings ...` to show where the warning was created)
> 2:44:12 PM [vite] Error when evaluating SSR module /src/lib/server/websearch/sentenceSimilarity.ts:
> |- TypeError: fetch failed
>     at fetch (/mnt/c/WSL/HuggingChat test/chat-ui/node_modules/undici/index.js:110:15)
>     at processTicksAndRejections (node:internal/process/task_queues:95:5)
>     at runNextTicks (node:internal/process/task_queues:64:3)
>     at listOnTimeout (node:internal/timers:540:9)
>     at process.processTimers (node:internal/timers:514:7)
>     at async getModelFile (file:///mnt/c/WSL/HuggingChat%20test/chat-ui/node_modules/@xenova/transformers/src/utils/hub.js:468:24)
>     at async getModelJSON (file:///mnt/c/WSL/HuggingChat%20test/chat-ui/node_modules/@xenova/transformers/src/utils/hub.js:542:18)
>     at async Promise.all (index 0)
>     at async loadTokenizer (file:///mnt/c/WSL/HuggingChat%20test/chat-ui/node_modules/@xenova/transformers/src/tokenizers.js:56:16)
>     at async AutoTokenizer.from_pretrained (file:///mnt/c/WSL/HuggingChat%20test/chat-ui/node_modules/@xenova/transformers/src/tokenizers.js:3778:48)
> 
> 2:44:12 PM [vite] Error when evaluating SSR module /src/lib/server/websearch/runWebSearch.ts: failed to import "/src/lib/server/websearch/sentenceSimilarity.ts"
> |- TypeError: fetch failed
>     at fetch (/mnt/c/WSL/HuggingChat test/chat-ui/node_modules/undici/index.js:110:15)
>     at processTicksAndRejections (node:internal/process/task_queues:95:5)
>     at runNextTicks (node:internal/process/task_queues:64:3)
>     at listOnTimeout (node:internal/timers:540:9)
>     at process.processTimers (node:internal/timers:514:7)
>     at async getModelFile (file:///mnt/c/WSL/HuggingChat%20test/chat-ui/node_modules/@xenova/transformers/src/utils/hub.js:468:24)
>     at async getModelJSON (file:///mnt/c/WSL/HuggingChat%20test/chat-ui/node_modules/@xenova/transformers/src/utils/hub.js:542:18)
>     at async Promise.all (index 0)
>     at async loadTokenizer (file:///mnt/c/WSL/HuggingChat%20test/chat-ui/node_modules/@xenova/transformers/src/tokenizers.js:56:16)
>     at async AutoTokenizer.from_pretrained (file:///mnt/c/WSL/HuggingChat%20test/chat-ui/node_modules/@xenova/transformers/src/tokenizers.js:3778:48)
> 
> 2:44:12 PM [vite] Error when evaluating SSR module /mnt/c/WSL/HuggingChat test/chat-ui/src/routes/conversation/[id]/+server.ts: failed to import "/src/lib/server/websearch/runWebSearch.ts"
> |- TypeError: fetch failed
>     at fetch (/mnt/c/WSL/HuggingChat test/chat-ui/node_modules/undici/index.js:110:15)
>     at processTicksAndRejections (node:internal/process/task_queues:95:5)
>     at runNextTicks (node:internal/process/task_queues:64:3)
>     at listOnTimeout (node:internal/timers:540:9)
>     at process.processTimers (node:internal/timers:514:7)
>     at async getModelFile (file:///mnt/c/WSL/HuggingChat%20test/chat-ui/node_modules/@xenova/transformers/src/utils/hub.js:468:24)
>     at async getModelJSON (file:///mnt/c/WSL/HuggingChat%20test/chat-ui/node_modules/@xenova/transformers/src/utils/hub.js:542:18)
>     at async Promise.all (index 0)
>     at async loadTokenizer (file:///mnt/c/WSL/HuggingChat%20test/chat-ui/node_modules/@xenova/transformers/src/tokenizers.js:56:16)
>     at async AutoTokenizer.from_pretrained (file:///mnt/c/WSL/HuggingChat%20test/chat-ui/node_modules/@xenova/transformers/src/tokenizers.js:3778:48)
> 
> TypeError: fetch failed
>     at fetch (/mnt/c/WSL/HuggingChat test/chat-ui/node_modules/undici/index.js:110:15)
>     at processTicksAndRejections (node:internal/process/task_queues:95:5)
>     at runNextTicks (node:internal/process/task_queues:64:3)
>     at listOnTimeout (node:internal/timers:540:9)
>     at process.processTimers (node:internal/timers:514:7)
>     at async getModelFile (file:///mnt/c/WSL/HuggingChat%20test/chat-ui/node_modules/@xenova/transformers/src/utils/hub.js:468:24)
>     at async getModelJSON (file:///mnt/c/WSL/HuggingChat%20test/chat-ui/node_modules/@xenova/transformers/src/utils/hub.js:542:18)
>     at async Promise.all (index 0)
>     at async loadTokenizer (file:///mnt/c/WSL/HuggingChat%20test/chat-ui/node_modules/@xenova/transformers/src/tokenizers.js:56:16)
>     at async AutoTokenizer.from_pretrained (file:///mnt/c/WSL/HuggingChat%20test/chat-ui/node_modules/@xenova/transformers/src/tokenizers.js:3778:48) {
>   cause: AggregateError
>       at internalConnectMultiple (node:net:1114:18)
>       at internalConnectMultiple (node:net:1177:5)
>       at internalConnectMultiple (node:net:1177:5)
>       at internalConnectMultiple (node:net:1177:5)
>       at internalConnectMultiple (node:net:1177:5)
>       at internalConnectMultiple (node:net:1177:5)
>       at Timeout.internalConnectMultipleTimeout (node:net:1687:3)
>       at listOnTimeout (node:internal/timers:575:11)
>       at process.processTimers (node:internal/timers:514:7) {
>     code: 'ETIMEDOUT',
>     [errors]: [
>       [Error], [Error],
>       [Error], [Error],
>       [Error], [Error],
>       [Error], [Error],
>       [Error], [Error],
>       [Error], [Error]
>     ]
>   }
> }
> (node:80446) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [TLSSocket]. Use emitter.setMaxListeners() to increase limit
> (node:80446) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [TLSSocket]. Use emitter.setMaxListeners() to increase limit
>

**Comment by nsarrazin** [comment]
This looks like a transformers.js fetching issue, maybe @xenova knows more about it than me. 👀 

I see a ` |- TypeError: fetch failed` in your trace, do you have some kind of proxy in place that would maybe block access to huggingface ?

**Comment by patchie** [comment]
> This looks like a transformers.js fetching issue, maybe @xenova knows more about it than me. 👀
> 
> I see a ` |- TypeError: fetch failed` in your trace, do you have some kind of proxy in place that would maybe block access to huggingface ?

ah, that might be the problem. I will investigate, thanks!

**Comment by Vonlife88** [comment]
> i don't think i was trying to do it of my ipad thought it was my email address or something

Sent with [Proton Mail](https://proton.me/) secure email.

**Comment by xenova** [comment]
This looks to be the same issue as https://github.com/huggingface/chat-ui/issues/528 ("Error when evaluating SSR module"). As discussed there, this module should not be rendered on the server-side.

Let me make a PR for the suggested change.

**Comment by nsarrazin** [comment]
The fix is merged @patchie can you give this another go ?

**Comment by xenova** [comment]
So, https://github.com/huggingface/chat-ui/pull/554 should fix this, but I don't know if you'll be able to use websearch, since it looks like `fetch` is disabled in your environment.

@patchie what version of node.js are you using? [This](https://stackoverflow.com/questions/49343024/getting-typeerror-failed-to-fetch-when-the-request-hasnt-actually-failed) might also be the issue.

Another possibility is that this is failing because transformers.js first looks to see if the models are present locally before fetching from the HF hub. If this is the case, we can maybe introduce an environment variable (or just update the default value) for `env.allowLocalModels=false`. Will wait for feedback before potentially opening up another PR.

**Comment by patchie** [comment]
> The fix is merged @patchie can you give this another go ?

Updated the code and executed the code from home(no proxy), Now it seems to work with both local model and external huggingface model inference.

I got this error in the terminal anyways, but not sure if it affected anything in the webUI:
![image](https://github.com/huggingface/chat-ui/assets/1792727/b3c49a1a-331a-413b-8873-279c79cd986d)

And i only got the error one time, when i'm chatting with local or remote inference models, it doesnt give me any more errors at all it seems.

PS, just as a note: When i start the dev-chat-ui, it takes a while for the website to load after the terminal shows me the url links. and when the website is loaded successfully and i start a chat, it is quite slow, and doesn't stream the inference. But after 5 min, both local and remote inference is super fast. Just wanted to tell you guys.


> @patchie what version of node.js are you using? 

Not sure if this is relevant anymore, but i am on "node -v" --> v20.9.0


> but I don't know if you'll be able to use websearch, since it looks like fetch is disabled in your environment.

I haven't enabled these 3 lines in the .env file, so i guess i will not be able to get the web search toggle for that reason.
```
YDC_API_KEY=#your docs.you.com api key here
SERPER_API_KEY=#your serper.dev api key here
SERPAPI_KEY=#your serpapi key here
```

Btw, super happy about the support here! tell me if i can do anything to help!
